### DISCUSSION TITLE
DeepSleep cycle hangs after many hours when  TelePeriod is 10

### DISCUSSION THREAD

**Discussion by Wirewrap** [arendst/Tasmota#14889] [Q&A] [answered]
I am designing a solar powered wind meter based on ESP2866 and Tasmota 10.1.
Tasmota is deep sleeping for 600 sec then wakes up, reads a speed count from a slave processor and reads wind direction from analog input. RESULT and TELE is sent.
If TelePeriod is set to 10 the process dies after several hours. If Teleperiod is set to 15 the meter can run forever, at least a week so far.
I understand "teleperiod == 10" is special, but how does it affect this case?

Power1 triggers the slave processor to send a serial message.
Power2 powers the wind direction potentiometer, it is then powered down during DeepSleep.

stat/Wind2/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":68,"Free":443,"Rules":"ON System#Boot do Backlog SerialSend2 xyz; power on; power off ENDON"}}
stat/Wind2/RESULT = {"Rule3":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":68,"Free":443,"Rules":"ON System#init DO power2 ON ENDON ON System#Save DO power2 off ENDON"}}

Attached cosole logs for TelePeriod 10 and 15

[Problem.txt](https://github.com/arendst/Tasmota/files/8088008/Problem.txt)

**Comment by ascillato2** [comment, marked as answer]
There are a lot of fixes from your version to latest from development branch. Please update and try again. Please, use the evelopment branch. Thanks.

**Reply by Wirewrap** [reply]
@ascillato2 Version 11.0.0.1 is up and running for more than 24 hrs now, so yes indeed!

**Comment by stefanbode** [comment]
@Wirewrap : very good question and we are just in the middle of some code changes that address your use case. First there was a small bug; fixed this morning where the wakeup was in the past. OK, I know bad ideas as long as there is no time travel. But this should only happen with small deepsleeptime e.g. 60. Longer times e.g. 3600 should not have the problem also in the past. 
The difference between teleperiod 10 and anything else is, that teleperiod 10 executes only ONCE and go to deepsleep. Any other teleperid will execute one time just after reboot and then a second time after teleperiod. This was added for users that want their device to stay awake for a little longer time. What is the POWER ON/OFF doing? we are just in the middle of fixing an issue where a sensor does not get initialized if the POWER is OFF on device start. Some timing issues we are just fixing.

**Comment by Wirewrap** [comment]
@stefanbode 
I use Power1 as a trigger for a slave processor to send a serial message to Tasmota.
Power2 is supply power for the wind direction potentiometer. I don't want to waste that power during DeepSleep

**Reply by stefanbode** [reply]
I was not sure if the serial message was correct and the device had enough time to send its message

**Reply by stefanbode** [reply]
power 1 is only ON for 0.3sec

**Reply by Wirewrap** [reply]
@stefanbode 
The negative edge of Power1 triggers the slave to send, Supply power to slave and ESP8266 is always on.
I think to use an I/O pin as logical trigger signal I have to give it a Power name?
The message is X{value}Y.  X and Y as delimiters.
